OneSignalSDK/onesignal/src/main/java/com/onesignal/OSNotificationController.java

```diff
@@ -0,0 +1,186 @@
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.app.NotificationCompat;
+
+import org.json.JSONObject;
+
+import static com.onesignal.OSUtils.isStringEmpty;
+
+public class OSNotificationController {
+
+   // The extension service app AndroidManifest.xml meta data tag key name
+   private static final String EXTENSION_SERVICE_META_DATA_TAG_NAME = "com.onesignal.NotificationServiceExtension";
+
+   private final OSNotificationGenerationJob notificationJob;
+   boolean isRestoring;
+   boolean isBackgroundLogic;
+
+   OSNotificationController(OSNotificationGenerationJob notificationJob, boolean isRestoring, boolean isBackgroundLogic) {
+      this.isRestoring = isRestoring;
+      this.isBackgroundLogic = isBackgroundLogic;
+      this.notificationJob = notificationJob;
+   }
+
+   OSNotificationController(Context context, JSONObject jsonPayload, boolean isRestoring, boolean isBackgroundLogic, Long timestamp) {
+      this.isRestoring = isRestoring;
+      this.isBackgroundLogic = isBackgroundLogic;
+
+      notificationJob = createNotificationJobFromCurrent(context, jsonPayload, timestamp);
+   }
+
+   /**
+    * Using current {@link OSNotificationController} class attributes, builds a {@link OSNotificationGenerationJob}
+    *    instance and returns it
+    * <br/><br/>
+    * @see OSNotificationGenerationJob
+    */
+   private OSNotificationGenerationJob createNotificationJobFromCurrent(Context context, JSONObject jsonPayload, Long timestamp) {
+      OSNotificationGenerationJob notificationJob = new OSNotificationGenerationJob(context);
+      notificationJob.setJsonPayload(jsonPayload);
+      notificationJob.setShownTimeStamp(timestamp);
+      notificationJob.setRestoring(isRestoring);
+      return notificationJob;
+   }
+
+   /**
+    * Called from {@link OSNotificationReceivedEvent#complete(OSNotification)} class
+    * If the notification modified by the user is null, the notification will be silent, otherwise will be displayed
+    * <br/><br/>
+    * @param originalNotification the notification received
+    * @param notification the notification sent by the user, might be modified
+    * @see OSNotificationReceivedEvent#complete(OSNotification)
+    */
+   void processNotification(OSNotification originalNotification, @Nullable OSNotification notification) {
+      OneSignal.onesignalLog(OneSignal.LOG_LEVEL.DEBUG, "processNotification called from Thread: " + Thread.currentThread().toString());
+      if (notification != null) {
+         // Save as processed to prevent possible duplicate calls from canonical ids
+         boolean display = isStringEmpty(notification.getBody());
+         if (!display) {
+            notDisplayNotificationLogic(originalNotification);
+         } else {
+            // Set modified notification
+            notificationJob.setNotification(notification);
+            NotificationBundleProcessor.processJobForDisplay(this, isBackgroundLogic);
```

Comment:
This is passing `isBackgroundLogic` but the param name for `processJobForDisplay` is `callForegroundLogic`.
Seems like we are flipping the logic here when we should not? Or just one of the names wrong?

Reply:
one name is wrong